packages/next-server/server/render.tsx

```diff
@@ -212,6 +212,10 @@ export async function renderToHTML(
       return render(renderElementToString, <ErrorDebug error={err} />)
     }
 
+    if ((props.router || props.Component) && ErrorDebug) {
+      throw new Error(`'router' and 'Component' can not be returned in getInitialProps from _app.js https://err.sh/zeit/next.js/cant-override-next-props.md`)
```

Comment:
Let's do console.warn here 👍 Or throw in development only.

Reply:
👌Updated